hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueFactory.kt

```diff
@@ -63,70 +67,61 @@ class HmppsQueueFactory(
-      sqsClient.createQueue(CreateQueueRequest(queueName))
+      sqsClient.createQueue(CreateQueueRequest.builder().queueName(queueName).build())
     } else {
-      sqsDlqClient.getQueueUrl(dlqName).queueUrl
-        .let { dlqQueueUrl -> sqsDlqClient.getQueueAttributes(dlqQueueUrl, listOf(QueueAttributeName.QueueArn.toString())).attributes["QueueArn"] }
-        .also { queueArn ->
-          sqsClient.createQueue(
-            CreateQueueRequest(queueName).withAttributes(
-              mapOf(
-                QueueAttributeName.RedrivePolicy.toString() to
-                  """{"deadLetterTargetArn":"$queueArn","maxReceiveCount":"5"}"""
-              )
-            )
-          )
-        }
+      val dlqUrl = sqsDlqClient.getQueueUrl(GetQueueUrlRequest.builder().queueName(dlqName).build()).queueUrl()
+      val dlqArn = sqsDlqClient.getQueueAttributes(GetQueueAttributesRequest.builder().queueUrl(dlqUrl).attributeNames(QueueAttributeName.QUEUE_ARN).build()).attributes()[QueueAttributeName.QUEUE_ARN]
+      sqsClient.createQueue(
+        CreateQueueRequest.builder().queueName(queueName).attributes(mapOf(QueueAttributeName.REDRIVE_POLICY to """{"deadLetterTargetArn":"$dlqArn","maxReceiveCount":"5"}""")).build()
+      )
     }
   }
 
   private fun subscribeToLocalStackTopic(hmppsSqsProperties: HmppsSqsProperties, queueConfig: QueueConfig, hmppsTopics: List<HmppsTopic>) {
-    if (hmppsSqsProperties.provider == "localstack")
+    if (findProvider(hmppsSqsProperties.provider) == Provider.LOCALSTACK) {
       hmppsTopics.firstOrNull { topic -> topic.id == queueConfig.subscribeTopicId }
         ?.also { topic ->
-          val subscribeAttribute = if (queueConfig.subscribeFilter.isNullOrEmpty()) mapOf() else mapOf("FilterPolicy" to queueConfig.subscribeFilter)
+          val subscribeAttribute = if (queueConfig.subscribeFilter.isEmpty()) mapOf() else mapOf("FilterPolicy" to queueConfig.subscribeFilter)
           topic.snsClient.subscribe(
-            SubscribeRequest()
-              .withTopicArn(topic.arn)
-              .withProtocol("sqs")
-              .withEndpoint("${hmppsSqsProperties.localstackUrl}/queue/${queueConfig.queueName}")
-              .withAttributes(subscribeAttribute)
+            SubscribeRequest.builder()
+              .topicArn(topic.arn)
+              .protocol("sqs")
+              .endpoint("${hmppsSqsProperties.localstackUrl}/queue/${queueConfig.queueName}")
+              .attributes(subscribeAttribute)
+              .build()
           )
             .also { log.info("Queue ${queueConfig.queueName} has subscribed to topic with arn ${topic.arn}") }
         }
+    }
   }
 
-  fun createJmsListenerContainerFactory(awsSqsClient: AmazonSQS, hmppsSqsProperties: HmppsSqsProperties): DefaultJmsListenerContainerFactory =
+  fun createJmsListenerContainerFactory(awsSqsClient: SqsClient, hmppsSqsProperties: HmppsSqsProperties): DefaultJmsListenerContainerFactory =
     DefaultJmsListenerContainerFactory().apply {
```

Comment:
JMS (as used in the [amazon-sqs-java-messaging-lib](https://github.com/awslabs/amazon-sqs-java-messaging-lib)) doesn't support async. As the async client no longer extends the sync client this has the consequence that we cannot create JMS listener factories for async clients - hence you won't find this funtion in `HmppsAsyncQueueFactory`. I imagine nearly all consumers of this library are using JMS and they'll all have to use the sync client!